File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/framework/httpapi/HttpApiLoader.java
package ru.usedesk.chat_sdk.internal.data.framework.httpapi;

import androidx.annotation.NonNull;

import java.io.IOException;

import javax.inject.Inject;

import okhttp3.ResponseBody;
import retrofit2.Response;
import ru.usedesk.chat_sdk.external.entity.UsedeskOfflineForm;
import ru.usedesk.common_sdk.internal.api.UsedeskApiFactory;

public class HttpApiLoader implements IHttpApiLoader {
    private final UsedeskApiFactory apiFactory;

    @Inject
    HttpApiLoader(@NonNull UsedeskApiFactory apiFactory) {
        this.apiFactory = apiFactory;
    }

    @Override
    public void post(@NonNull String baseUrl, @NonNull UsedeskOfflineForm offlineForm) throws IOException {
        IHttpApi httpApi = apiFactory.getInstance(baseUrl, IHttpApi.class);
        Response<ResponseBody> response = httpApi.postOfflineForm(offlineForm).execute();

        if (response.isSuccessful() && response.body() != null) {
            return;
        }
        throw new IOException("Server error: " + response.code());
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/framework/httpapi/HttpApiLoader.kt
package ru.usedesk.chat_sdk.internal.data.framework.httpapi

import ru.usedesk.chat_sdk.external.entity.old.UsedeskOfflineForm
import ru.usedesk.chat_sdk.internal.data.framework._extra.retrofit.IHttpApiFactory
import toothpick.InjectConstructor
import java.io.IOException

@InjectConstructor
class HttpApiLoader(
        private val httpApiFactory: IHttpApiFactory
) : IHttpApiLoader {

    @Throws(IOException::class)
    override fun post(baseUrl: String, offlineForm: UsedeskOfflineForm) {
        val httpApi = httpApiFactory.getInstance(baseUrl)
        val response = httpApi.postOfflineForm(offlineForm).execute()
        if (response.isSuccessful && response.body() != null) {
            return
        }
        throw IOException("Server error: " + response.code())
    }
}